src/lib/core/CASEAuthTag.h

```diff
@@ -84,6 +87,16 @@ struct CATValues
         }
         return CHIP_NO_ERROR;
     }
+
+private:
+    static bool Cmp(const CASEAuthTag & a, const CASEAuthTag & b) { return a < b; }
+    void Sort() { Sorting::BubbleSort(this->values.begin(), this->size(), Cmp); }
+    CATValues Sorted() const
+    {
+        auto sorted = *this;
+        sorted.Sort();
+        return sorted;
```

Comment:
How frequently must this be done?

It's not "super" expensive, but it's expensive if we do it all the time.

std::sort() and quicksort() have special casing in most implementations that uses sorting networks for small arrays (e.g. arrays of 3 values), which is very and costs very little since already used in other parts of the system as well (usually)

Reply:
Yes, normally I would prefer to use std::sort or qsort.  But std::sort() was explicitly rejected in another PR, and I think it was shown in the increases report that it was being used for the first time in many builds when it was considered for introduction.  So bubble sort was added to our libs, and I am using that to avoid any increase in flash.  I won't argue what is optimal.  I am following the result of the prior discussion.

However, for performance, I think the big problem here is that: 1) we need to sort every time we need one of these operators, and 2) we need to push 192 bytes on the stack to do it.  Because we can't sort in place.

I will fix those two problems.  Actually this seems related to your other comment here: https://github.com/project-chip/connectedhomeip/pull/20253#discussion_r912555429